pipelines/batch/src/main/java/com/google/fhir/analytics/JdbcFetchHapi.java

```diff
@@ -135,34 +137,78 @@ public ResourceTag mapRow(ResultSet resultSet) throws Exception {
     }
   }
 
+  public static class MdmLinkToRowDescriptor implements JdbcIO.RowMapper<MdmLink> {
+
+    @Override
+    public MdmLink mapRow(ResultSet resultSet) throws Exception {
+      return MdmLink.builder()
+          .sourceFhirId(resultSet.getString("source_fhir_id"))
+          .goldenFhirId(resultSet.getString("golden_fhir_id"))
+          .resourceId(resultSet.getString("src_resource_id"))
+          .build();
+    }
+  }
+
+  public static class SourceIdentifierToRowDescriptor
+      implements JdbcIO.RowMapper<SourceIdentifier> {
+
+    @Override
+    public SourceIdentifier mapRow(ResultSet resultSet) throws Exception {
+      return SourceIdentifier.builder()
+          .resourceId(resultSet.getString("golden_resource_pid"))
+          .system(resultSet.getString("sp_system"))
+          .value(resultSet.getString("sp_value"))
+          .build();
+    }
+  }
+
   /**
    * Utilizes Beam JdbcIO to query for resources directly from FHIR (HAPI) server's database and
    * returns a PCollection of Lists of String objects - each corresponding to a resource's payload
    */
   public static class FetchRowsJdbcIo
       extends PTransform<PCollection<QueryParameterDescriptor>, PCollection<HapiRowDescriptor>> {
 
-    private String resourceList;
+    private final String resourceList;
     private final JdbcIO.DataSourceConfiguration dataSourceConfig;
 
     private final String query;
 
     private final String tagQuery;
 
+    private final String mdmQuery;
+
+    private final boolean isMdmResource;
+    private final boolean mapToGoldenResources;
+    private String identifiersQuery;
```

Comment:
**question:** All the rest of the fields in the class are `final`, should this one be as well?